modules/client/test/unit/context.test.ts

```diff
@@ -27,54 +36,119 @@ describe("Context instances", () => {
+    expect(context.web3Providers.length).toBe(WEB3_NODES.homestead.length);
+    expect(context.ipfs.length).toBe(IPFS_NODES.homestead.length);
+    expect(context.graphql.length).toBe(GRAPHQL_NODES.homestead.length);
+    context.web3Providers.map((provider) => {
+      expect(provider).toBeInstanceOf(JsonRpcProvider);
+    });
+    context.ipfs.map((ipfsClient) => {
+      expect(ipfsClient).toBeInstanceOf(IpfsClient);
+    });
+    context.graphql.map((graphqlClient) =>
+      expect(graphqlClient).toBeInstanceOf(GraphQLClient)
+    );
   });
   it("Should create a context and have the correct values", () => {
     const context = new Context(contextParams);
 
     expect(context).toBeInstanceOf(Context);
-    expect(context.network).toEqual("mainnet");
-    expect(context.signer).toBeInstanceOf(Wallet);
-    expect(context.daoFactoryAddress).toEqual("0x1234");
-    context.web3Providers?.map((provider) =>
+    expect(context.network.name).toBe("homestead");
+    expect(context.network.chainId).toBe(1);
+    expect(context.daoFactoryAddress).toBe(contextParams.daoFactoryAddress);
+    expect(context.ensRegistryAddress).toBe(context.network.ensAddress);
+    expect(context.gasFeeEstimationFactor).toBe(
+      contextParams.gasFeeEstimationFactor,
+    );
+    context.web3Providers.map((provider) =>
       expect(provider).toBeInstanceOf(JsonRpcProvider)
     );
-    expect(context.gasFeeEstimationFactor).toEqual(0.1);
+    context.ipfs.map((ipfsClient) =>
+      expect(ipfsClient).toBeInstanceOf(IpfsClient)
+    );
+    context.graphql.map((graphqlClient) =>
+      expect(graphqlClient).toBeInstanceOf(GraphQLClient)
+    );
   });
   it("Should set a new context and have the correct values", () => {
     const context = new Context(contextParams);
     contextParams = {
-      network: "mainnet",
+      network: "goerli",
       signer: new Wallet(TEST_WALLET),
       daoFactoryAddress: "0x2345",
       web3Providers: web3endpoints.working,
       gasFeeEstimationFactor: 0.1,
       ipfsNodes: [{ url: "https://localhost", headers: {} }],
       graphqlNodes: [{ url: "https://localhost" }],
     };
-    context.setFull(contextParams);
+    context.set(contextParams);
 
     expect(context).toBeInstanceOf(Context);
-    expect(context.network).toEqual("mainnet");
+    expect(context.network.name).toEqual("goerli");
+    expect(context.network.chainId).toEqual(5);
     expect(context.signer).toBeInstanceOf(Wallet);
     expect(context.daoFactoryAddress).toEqual("0x2345");
     context.web3Providers?.map((provider) =>
       expect(provider).toBeInstanceOf(JsonRpcProvider)
     );
+    context.ipfs?.map((ipfsClient) =>
+      expect(ipfsClient).toBeInstanceOf(IpfsClient)
+    );
+    context.graphql?.map((graphqlClient) =>
+      expect(graphqlClient).toBeInstanceOf(GraphQLClient)
+    );
     expect(context.gasFeeEstimationFactor).toEqual(0.1);
   });
+  it("Should Change the network and update all the parameters", () => {
```

Comment:
I'm missing certain test cases:

- Create a context with empty params. Set the network to mumbai and see that everything changed
- Create a context by passing only the network (goerli). See that all params are from development. Then set the network to matic. See that everything changed again

Reply:
ping

Reply:
Where comes again the question whether an empty context should be possible, or rather having some mandatory parameters

Reply:
Yes, check previous comment, but if we add mandatory params I'd add `Signer` and `web3Providers`

keeping them non mandatory can be useful in some UI cases when you whant to have initialized the client, but you dont have a signer because the user is not logged in. For example aragon APP landing page where it wheries the sdk for showing daos and proposals but the signer and web3providers can be undefined.

Reply:
I see, makes sense. Then, do we have checks that make the Client fail whenever a Context without the required fields is passed?